### ISSUE TITLE
Add option to ignore deprecation inside a deprecated context

### ISSUE THREAD

**Issue by RebeccaStevens** [gund/eslint-plugin-deprecation#56]
For example, this would not cause anything to be flagged:

```ts
/**
 * @deprecated
 */
function foo() {
  call_to_deprecated_function();
}
```

But if `@deprecated` is removed from `foo`, then `call_to_deprecated_function` is flagged.

If I get some free time I might look into making a PR for this.

**Comment by gund** [comment]
I'm not sure it should not report deprecated usage within deprecated code as you still should be able too see every place where your code accesses deprecated things.

Also even if you would skip `call_to_deprecated_function` call what should happen if within that function it accesses another deprecated API, should it also skip that or report as deprecated?
If skip - it will introduce complicated logic which has to track call stacks, if not - it will become inconsistent at some place it will be reported and at another place it will not depending on the docblocks.

I would recommend simply using eslint native rule disabling comments for cases such as you reported and not introduce extra logic to an already complicated and [quite slow](https://github.com/gund/eslint-plugin-deprecation/issues/44) rule.
